src/algebra/regular/smul.lean

```diff
@@ -56,17 +59,18 @@ is `M`-regular. -/
   is_smul_regular M (a • b) ↔ is_smul_regular M b :=
 ⟨of_smul _, ha.smul⟩
 
+lemma is_left_regular_iff [has_mul R] {a : R} :
+  is_left_regular a ↔ is_smul_regular R a := iff.rfl
```

Comment:
Generalizing from `mul_action` to `monoid` is nice, but please don't throw away the docstring!

Also, let's move this to the root namespace, since `is_smul_regular.is_left_regular_iff` is kind of a weird name.
```suggestion
/-- Left-regular multiplication on `R` is equivalent to `R`-regularity of `R` itself. -/
lemma _root_.is_left_regular_iff [has_mul R] {a : R} :
  is_left_regular a ↔ is_smul_regular R a := iff.rfl
```

Reply:
Can you add the same lemma for `is_right_regular_iff`, as `is_right_regular a ↔ is_smul_regular R (op a)`?